tests/attribute_definition_test.py

```diff
@@ -92,7 +92,7 @@ def test_simple_data_verifier_failure():
                                "<class 'bool'>, but <class 'int'> given.")
 
 
-class IntListMissingVerifierData(Data[List[int]]):
+class IntListMissingVerifierData(Data[list[int]]):
```

Comment:
It seems that this change here is crashing the test. What seems to happen is that Data[List[int]] is not a class, whereas Data[list[int]] is one. This makes it s.t. in the irdl file, line 901 is not triggered and the expection is not raised.

 I am honestly not sure about this and how this should be handled. Maybe we need to change that check? I guess @math-fehr has the best guess at what to do.

Reply:
I see what is happening here.
It is said "The definition should fail, since no verifier is provided, and the Data type parameter is not a class."
The second part failed due the change. How about the first part? It seems that a verifier has been generated for list of int?

Reply:
From comparing to the version on main, I think the first part is intended.

Reply:
Here is what I think happened.

Currently, if you define a `Data` with something that is not a class, you need to provide a verifier to check that the parameter is of the right type. The way you check if you have the right type is by using `isinstance`. However, for generic types, you can't just use `isinstance`, since they aren't classes, but `GenericAlias`.

Now for the fun thing:
My assumption was that if you are a class, then you can use `isinstance`, and the world is good. However, it seems that, for `list[int]` this is not the case, since you get the error `isinstance() argument 2 cannot be a parameterized generic`.
I think that since `list[int]` is a subclass of `GenericAlias' (https://peps.python.org/pep-0585/), then we should just check that the parameter is a class AND is not a subclass of `GenericAlias`. Hopefully, that should work!